cluster/autoscale.go

```diff
@@ -146,3 +156,101 @@ func createAutoscalingForAzure(cluster CommonCluster, groups []nodeGroup) *autos
 		},
 	}
 }
+
+//DeployClusterAutoscaler post hook only for AWS & Azure for now
+func DeployClusterAutoscaler(cluster CommonCluster) error {
+
+	var nodeGroups []nodeGroup
+
+	switch cluster.GetType() {
+	case banzaiConst.Amazon:
+		nodeGroups = getAmazonNodeGroups(cluster)
+	case banzaiConst.Azure:
+		nodeGroups = getAzureNodeGroups(cluster)
+	default:
+		return nil
```

Comment:
I'd error out with a not supported message here.

Reply:
Yes but calling with other providers is not really an error, this like more like we deploy autoscaler only for Amazon & Azure. I could make this check before calling DeployClusterAutoscaler, but then I had to check this all the places this function is invoked.